terraform/lib/dependabot/terraform/file_parser.rb

```diff
@@ -49,29 +48,43 @@ def parse_terraform_file(file)
       end
 
       def parse_terragrunt_file(file)
-        modules = parsed_file(file).fetch("terraform", []).first || {}
-        modules.each do |details|
-          next unless details.is_a?(Hash)
-          next unless details.key?("terraform")
-            @dependency_set << build_terragrunt_dependency(file, details)
+        modules = parsed_file(file).fetch("terraform", []).
+                  map { |k, v| [k, v] }.to_h
+        modules.each do |name, details|
+          next unless name == "source"
+
+          @dependency_set << build_terraform_dependency(
+            file, "", [Hash[name, details]]
+          )
         end
       end
 
       def parse_terragrunt_legacy_file(file)
-        modules = parsed_file(file).fetch("terragrunt", []).first || {}
-        modules = modules.fetch("terraform", [])
-        modules.each do |details|
-          next unless details["source"]
+        modules = parsed_file(file).fetch("terraform", []).
```

Comment:
Is this function now exactly the same as `parse_terragrunt_file()`? If so, on what is currently line 33, can we instead just call `parse_terragrunt_file()`, then remove `parse_terragrunt_legacy_file()`?

Reply:
It is the same, but I wasn't sure that differences wouldn't be discovered when the 'unit tests' (spec files) were augmented.  The fact that the terragrunt and terragrunt legacy processing is the same seems suspicious to me and I wondered upon further research that there might be differences.  But this pull requests gets things moving along and that research could be postponed until work is done on the 'unit tests'.

However, If you are comfortable with the fact that there shouldn't be differences or differences that we need to handle, then I can make the changes you suggest.  Or, I can make the changes you suggest now and come back later and separate the two functions.

Reply:
Ahh, well the main differences I think are that the legacy syntax is hcl1, and they overloaded the terraform.tfvars concept of terraform to also contain the terragrunt configuration. The new syntax is hcl2, and they have a dedicated configuration file completely separate from terraform.

In the old style, the file is `terraform.tfvars` and the top-level key for the terragrunt config is `terragrunt`. Nested within `terragrunt` is the key `terraform`.

In the new style, the file is `terragrunt.hcl` and there is no `terragrunt` key because the file is dedicated. So the relevant top-level key is `terraform`. Within this section, the `source` is the same as the legacy style.

Looking at the changes again, I don't think `parse_terragrunt_legacy_file()` will work since it is fetching the key `terraform` from the parsed file, instead of `terragrunt`...?

Reply:
1.  Based on what you said, then I think `parse_terragrunt_legacy_file` would need to nest two levels.  I'll create a test file and try it out.  I'll make any necessary code changes.
2.  I did a github search through the Plus3IT repos (`org:plus3it filename:terraform.tfvars`) and 5 files were found, and only one (`terraform-external-s3-keys`) mentioned `terragrunt`.  That repo has a `terraform.tfvars` file that simply consists of `terragrunt {}`.    It's a "sample" file.  So no immediate impact there.
3.  The "unit tests" are not adequate.  I'll add another note to the Trello task to make sure we test a legacy terragrunt file with a source key.

Reply:
There is a test fixture already for `terraform.tfvars`, but it's using that git/ssh syntax that was giving you trouble: https://github.com/plus3it/dependabot-core/blob/tf012/terraform/spec/fixtures/config_files/terragrunt.tfvars

Looking into that, might be because that repo no longer exists and doesn't even redirect. Fix could be as simple as changing the source to: `git::git@github.com:gruntwork-io/terragrunt-infrastructure-modules-example.git//mysql?ref=v0.0.2`

Reply:
When I skimmed the tests, it appeared the ssh form was only in use when testing the source repo dependency.  So there are other instances.  I also want to make sure that the change I'm making now has adequate tests.  Plus there are two failing tests and one of them I really didn't look into yet.  Would you like me to stop working on the IAM Role task and instead work on the dependabot tests?

Reply:
Changes made to correct terragrunt legacy file.  I added some more test files to my test-dependabot repo, but made no changes to the unit tests at this time.